src/drop-down/drop-down-item.component.ts

```diff
@@ -0,0 +1,167 @@
+import {
+    Component,
+    ElementRef,
+    forwardRef,
+    HostBinding,
+    HostListener,
+    Inject,
+    Input } from "@angular/core";
+import { IgxDropDownComponent, ISelectionEventArgs } from "./drop-down.component";
+
+/**
+ * The `<igx-drop-down-item> is a container intended for row items in
+ * a `<igx-drop-down>` container.
+ */
+@Component({
+    selector: "igx-drop-down-item",
+    templateUrl: "drop-down-item.component.html"
+})
+
+export class IgxDropDownItemComponent {
+    private _isFocused = false;
+
+    @HostBinding("class.igx-drop-down__item")
+    get itemStyle(): boolean {
+        return !this.isHeader;
+    }
+
+    /**
+     * Gets if the item is the currently selected one in the dropdown
+     */
+    get isSelected() {
+        return this.dropDown.selectedItem === this;
+    }
+
+    @HostBinding("attr.aria-selected")
+    @HostBinding("class.igx-drop-down__item--selected")
+    get selectedStyle(): boolean {
+        return this.isSelected;
+    }
+
+    /**
+     * Sets/gets if the given item is focused
+     */
+    @HostBinding("class.igx-drop-down__item--focused")
+    get isFocused() {
+        return this._isFocused;
+    }
+    set isFocused(value: boolean) {
+        if (this.isDisabled || this.isHeader) {
+            this._isFocused = false;
+            return;
+        }
+
+        if (value && !this.dropDown.toggleDirective.collapsed) {
+            this.elementRef.nativeElement.focus();
+        }
+        this._isFocused = value;
+    }
+
+    /**
+     * Sets/gets if the given item is header
+     */
+    @Input()
+    @HostBinding("class.igx-drop-down__header")
+    public isHeader = false;
+
+    /**
+     * Sets/gets if the given item is disabled
+     */
+    @Input()
+    @HostBinding("class.igx-drop-down__item--disabled")
+    public isDisabled = false;
+
+    @HostBinding("attr.tabindex")
+    get setTabIndex() {
+        const shouldSetTabIndex = this.dropDown.allowItemsFocus && !(this.isDisabled || this.isHeader);
+        console.log(shouldSetTabIndex);
```

Comment:
Should we not include console.log calls in production code?

Reply:
We should not :)
Done.